internal/resourcepolicies/resource_policies.go

```diff
@@ -78,26 +85,49 @@ func (policies *Policies) buildPolicy(resPolicies *resourcePolicies) error {
 	return nil
 }
 
-func (p *Policies) Match(res interface{}) *Action {
-	vol, ok := res.(*StructuredVolume)
-	if ok {
-		for _, policy := range p.VolumePolicies {
-			isAllMatch := false
-			for _, con := range policy.conditions {
-				if !con.Match(vol) {
-					isAllMatch = false
-					break
-				}
-				isAllMatch = true
-			}
-			if isAllMatch {
-				return &policy.action
+func (p *Policies) match(res *structuredVolume) *Action {
+	for _, policy := range p.VolumePolicies {
+		isAllMatch := false
+		for _, con := range policy.conditions {
+			if !con.Match(res) {
+				isAllMatch = false
+				break
 			}
+			isAllMatch = true
+		}
+		if isAllMatch {
+			return &policy.action
 		}
 	}
 	return nil
 }
 
+func (p *Policies) MatchResourcePolicies(res interface{}, resType ResourceType) (*Action, error) {
```

Comment:
1. `MatchResourcePolicies` is a little repetitive, you may just call it `MatchResource` if `Match` is unclear.
2. Doesn't `res` contain the information of `resType`?

Reply:
res is one interface type we need to try all possible type conventions.

Add `resType` as I want the type convention precisely, may it not be graceful in function, but it could reduce the type conversions frequency.

Reply:
In fact, it returns an Action by inputting one resource, What about `MatchAction` or `GetMatchAction` is more suitable?

Reply:
How about transform it to unstructured object first and do the transformation based on`GetObjectKind`?

Reply:
it will make the function more graceful, but we call the `MatchResourcePolicies` we already have the structured object, we need unstructured when calling the function, and structured it again inner. Maybe it's not effecient

Reply:
I've removed `resType`